src/api/app/controllers/webui/package_controller.rb

```diff
@@ -92,22 +92,25 @@ def linking_packages
   end
 
   def dependency
-    unless Project.find_by_name(params[:dproject].to_s)
+    dep_project = Project.find_by_name(params[:dproject]) || Project.find_remote_project(params[:dproject]).try(:first)
+    unless dep_project
       flash[:error] = "Project '#{params[:dproject]}' is invalid."
       redirect_back(fallback_location: root_path)
       return
     end
 
     unless Architecture.archcache.include?(params[:arch])
       flash[:error] = "Architecture '#{params[:arch]}' is invalid."
-      redirect_back(fallback_location: project_show_path(project: params[:dproject]))
+      redirect_back(fallback_location: project_show_path(project: @project.name))
```

Comment:
Why does this change?

Reply:
`@project` refers to the project of the package, so this is where we want to redirect back the user if something goes wrong. With `params[:dproject]`, it wasn't necessary the project of the package. It can refer to a remote project for example, for which we cannot redirect back to. One example, you have the interconnect for openSUSE.org and one package is contained in a project linking to `openSUSE.org:Factory`, then the redirection would try to redirect the user back to `openSUSE.org:Factory`, which is not even there in the current instance (only on the remote one)